include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
     D3D12_NODE_MASK m_Desc;
 };
 
+//------------------------------------------------------------------------------------------------
+// Peter: Implementation for the new CheckFeatureSupport API
+
+#define FEATURE_SUPPORT_GET(RETTYPE,FEATURE,OPTION) \

```

Comment:
Macros to set up an getter function for each entry in the feature data. Right now the getter function has the same name as the feature entry (if the entry is called "SomeFeatureSupported" then the getter function is also "SomeFeatureSupported()").

Reply:
Good use of the macros :) The GitHub comments were a great help understanding what each one did without having to see them in use. Thoughts on adding them as comments in the code too?

Reply:
Oops, looks like the page hadn't loaded Jesse's comments before I replied. I'll defer to him here.

Reply:
I'm using the macros so that if I need to change how similar functions are implemented, I can change it in one place instead of going over the code and changing all of them. This may save some time in the development phase, but indeed it's achieved at the cost of code readability. I'll probably keep the macros until the design patterns are finalized, and by that time I'll start replacing unnecessary macros with code. Does that sounds good?

Reply:
I'll add the comments on what these macros do to the code as well. Thanks for the advice!